pkg/scalers/splunk_o11y_scaler.go

```diff
@@ -0,0 +1,264 @@
+		signalflow.AccessToken(accessToken))
+	if err != nil {
+		return nil, fmt.Errorf("error creating SignalFlow client: %w", err)
+	}
+
+	return apiClient, nil
+}
+
+func logMessage(logger logr.Logger, msg string, value float64) {
+	if value != -1 {
+		msg = fmt.Sprintf("splunk_o11y_scaler: %s -> %v", msg, value)
+	} else {
+		msg = fmt.Sprintf("splunk_o11y_scaler: %s", msg)
+	}
+	logger.Info(msg)
+}
+
+func (s *splunkO11yScaler) getQueryResult(ctx context.Context) (float64, error) {
+	// var duration time.Duration = 1000000000 // one second in nano seconds
+	var duration time.Duration = 10000000000 // ten seconds in nano seconds
+
+	comp, err := s.apiClient.Execute(context.Background(), &signalflow.ExecuteRequest{
+		Program: s.metadata.query,
+	})
+	if err != nil {
+		return -1, fmt.Errorf("error: could not execute signalflow query: %w", err)
+	}
+
+	go func() {
+		time.Sleep(duration)
+		if err := comp.Stop(context.Background()); err != nil {
+			s.logger.Info("Failed to stop computation")
+		}
+	}()
+
+	logMessage(s.logger, "Received Splunk Observability metrics", -1)
+
+	max := math.Inf(-1)
+	min := math.Inf(1)
+	valueSum := 0.0
+	valueCount := 0
+	s.logger.Info("getQueryResult -> Now Iterating")
+	for msg := range comp.Data() {
+		s.logger.Info("getQueryResult -> msg: %+v\n", msg)
+		if len(msg.Payloads) == 0 {
+			s.logger.Info("getQueryResult -> No data retreived. Continuing")
+			continue
+		}
+		for _, pl := range msg.Payloads {
+			value, ok := pl.Value().(float64)
+			if !ok {
+				return -1, fmt.Errorf("error: could not convert Splunk Observability metric value to float64")
+			}
+			logMessage(s.logger, "Encountering value ", value)
+			if value > max {
+				max = value
+			}
+			if value < min {
+				min = value
+			}
+			valueSum += value
+			valueCount++
+		}
+	}
+
+	if valueCount > 1 && s.metadata.queryAggegrator == "" {
+		return 0, fmt.Errorf("query returned more than 1 series; modify the query to return only 1 series or add a queryAggregator")
+	}
+
+	switch s.metadata.queryAggegrator {
+	case "max":
+		logMessage(s.logger, "Returning max value ", max)
+		return max, nil
+	case "min":
+		logMessage(s.logger, "Returning min value ", min)
+		return min, nil
+	case "avg":
+		avg := valueSum / float64(valueCount)
+		logMessage(s.logger, "Returning avg value ", avg)
+		return avg, nil
+	default:
+		return max, nil
+	}
+}
+
+func (s *splunkO11yScaler) GetMetricsAndActivity(ctx context.Context, metricName string) ([]external_metrics.ExternalMetricValue, bool, error) {
+	// s.logger.Info(fmt.Sprintf("splunk_o11y_scaler found authtrigger token : %s", s.metadata.accessToken))
+	num, err := s.getQueryResult(ctx)
+
+	if err != nil {
+		s.logger.Error(err, "error getting metrics from Splunk Observability Cloud.")
+		return []external_metrics.ExternalMetricValue{}, false, fmt.Errorf("error getting metrics from Splunk Observability Cloud: %w", err)
+	}
+	metric := GenerateMetricInMili(metricName, num)
+
+	logMessage(s.logger, "num", num)
```

Comment:
`num` will never equal -1 here (-1 is returned from `s.getQueryResult()` only when error is returned and such a scenario id handled in line 237).